src/main.rs

```diff
@@ -158,7 +160,67 @@ pub fn ask_to_run(mut cmd: Command, ask: bool, text: &str) {
     }
 }
 
-fn build_sysroot(auto: bool, target: &str, rustc_version: &VersionMeta) -> PathBuf {
+/// Returns the sanitizer to use.
+/// Ok(Some(str)) => Use sanitizer "string"
+/// Ok(None) => Use no sanitizer
+///
+/// # Errors
+/// Err(err) => There was an error when trying to get the list of supported sanitizers
+pub fn sanitizer(target: &str) -> anyhow::Result<Option<&'static str>> {
+    // To get the list of supported sanitizers, we call `rustc --print target-spec-json`
+    // and parse the output.
+
+    let mut cmd = rustc();
+    cmd.args([
+        "-Z",
+        "unstable-options",
+        "--print",
+        "target-spec-json",
+        "--target",
+        target,
+    ]);
+    let output = match cmd.output() {
```

Comment:
The pattern I usually use is `cmd.output().context("failed to ...")?`. That's much less noisy than explicitly matching. Any reason not to do the same here?